src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -101,6 +100,17 @@ internal static class ParameterHelpers
             return parameters;
         }
 
+        internal static void EnsureIsReadOnlyAttributeExists(IEnumerable<ParameterSymbol> parameters, DiagnosticBag diagnostics, bool modifyCompilationForRefReadOnly)
+        {
+            foreach (var parameter in parameters)
+            {
+                if (parameter.RefKind == RefKind.RefReadOnly)
+                {
+                    parameter.DeclaringCompilation.EnsureIsReadOnlyAttributeExists(diagnostics, parameter.GetNonNullSyntaxNode().Location, modifyCompilationForRefReadOnly);
+                }
+            }
+        }
```

Comment:
Unfortunately no, because `MakeParameters` is sometimes called during symbol creation. We want to do that during binding (to make sure we can do lookups without causing circular binding SOE).